rebuild/src/addon.cc

```diff
@@ -0,0 +1,90 @@
+#include "addon.h"
+
+bool is_zero_bytes(
+    const uint8_t *data, const size_t start_byte, const size_t byte_length) {
+    for (size_t i = start_byte; i < byte_length; i++) {
+        if (data[i] != 0) {
+            return false;
+        }
+    }
+    return true;
```

Comment:
A more idiomatic C++ code would be along the lines of `return std::all_of(data + start_byte, data + byte_length, [](auto x) { x == 0x00 })`. It also might be worth checking for bounds at the beginning, i.e., if `start_byte < byte_length`.

Reply:
nice!  will make that change! thanks!

Reply:
updated to

```c++
bool is_zero_bytes(
    const uint8_t *data,
    const size_t start_byte,
    const size_t byte_length) noexcept {
    if (start_byte < byte_length) {
        return std::all_of(
            data + start_byte, data + byte_length, [](uint8_t x) {
                return x == 0x00;
            });
    }
    return false;
}
```